plugins/by-name/conform-nvim/default.nix

```diff
@@ -234,4 +259,18 @@ lib.nixvim.plugins.mkNeovimPlugin {
       }
     ```
   '';
+
+  extraConfig =
+    cfg:
+    let
+      inherit (import ./auto-install.nix { inherit pkgs lib; }) getPkgFromConformName collectFormatters;
+      getPkgFromConformName' = getPkgFromConformName {
+        customFormatters = mapAttrs (_: _: null) cfg.settings.formatters;
+        inherit (cfg.autoInstall) overrides;
+      };
+      names = collectFormatters (attrValues cfg.settings.formatters_by_ft);
+    in
+    {
+      extraPackages = lib.optionals cfg.autoInstall.enable (map getPkgFromConformName' names);
```

Comment:
`lib.mkIf` and `mkMerge` are almost always the better approach when creating module-system option definitions.

The difference is that creating an "`optional`" definition will _always_ create a definition, either of `[]` or the actual value (`map ... names`), while using `mkIf` will only create the definition if the condition is true.

This comes into play when deciding which "override" priority to use, and when merging multiple definitions. E.g. a `mkIf false (mkForce "")` definition will not actually create the `mkForce ""` definition, so the highest override priority will likely remain lower than "force".